pandas/io/common.py

```diff
@@ -186,16 +190,39 @@ def get_filepath_or_buffer(filepath_or_buffer, encoding=None,
     ----------
     filepath_or_buffer : a url, filepath (str, py.path.local or pathlib.Path),
                          or buffer
+            now supports 'https://<user>:<password>@<host>:<port>/<url-path>'
+
+            .. versionadded:: 0.21.0
+
     encoding : the encoding to use to decode py3 bytes, default is 'utf-8'
 
+    compression : string, default None
+
+            .. versionadded:: 0.18.1
+
+    auth : tuple, default None
+            A tuple of string with (username, password) string for
+            HTTP(s) basic auth: eg auth= ('roberto', 'panda$4life')
+
+            .. versionadded:: 0.21.0
+
+    verify_ssl : boolean, Default True
```

Comment:
I think this is an attempt to mirror what `requests` does, which is to check SSL certificates by default.  I for one second that defaulting.